openhtf/io/user_input.py

```diff
@@ -133,20 +143,27 @@ def Stop(self):
 
   def run(self):
     """Main logic for this thread to execute."""
-    # First, display the prompt to the console.
-    print self._message
-
-    # Before reading, clear any lingering buffered terminal input.
-    termios.tcflush(sys.stdin, termios.TCIOFLUSH)
-
-    while not self._stopped:
-      inputs, _, _ = select.select([sys.stdin], [], [], 0.001)
-      for stream in inputs:
-        if stream == sys.stdin:
-          response = sys.stdin.readline().rstrip()
-          self._callback(response)
-          self._stopped = True
-          return
+    try:
+      if platform.system() == 'Windows':
+        # Windows doesn't support file-like objects for select(), so fall back
+        # to raw_input().
+        self._callback(raw_input(self._message + '\n\r'))
+      else:
+        # First, display the prompt to the console.
+        print self._message
+
+        # Before reading, clear any lingering buffered terminal input.
+        termios.tcflush(sys.stdin, termios.TCIOFLUSH)
+
+        while not self._stopped:
+          inputs, _, _ = select.select([sys.stdin], [], [], 0.001)
+          for stream in inputs:
+            if stream is sys.stdin:
+              response = sys.stdin.readline().rstrip()
+              self._callback(response)
+              return
+    finally:
+      self._stopped = True
```

Comment:
i'm guessing it's because Stop() can be called multiple times from different threads.  I put it here for consistency with the previous implementation.  we can clean this up in a separate PR if you want, for now i'd rather just leave the same behavior as it had.